tests/TestCase/Rector/MethodCall/OptionsArrayToNamedParametersRector/Fixture/options_to_named_parameters.php.inc

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace Cake\Upgrade\Test\TestCase\Rector\MethodCall\OptionsArrayToNamedParametersRector\Fixture;
+
+use Cake\Upgrade\Rector\Tests\Rector\MethodCall\OptionsArrayToNamedParametersRector\Source;
+
+function optionsToNamedParameters()
+{
+    $value = 1;
+    $instance = new Source\ConfigurableClass();
+
+    $instance->find();
+    $instance->find('all');
+    $instance->find('list', ['fields' => ['name']]);
+    $instance->find('all', [
+        'conditions' => ['Articles.id' => $value],
+        'order' => ['Articles.id' => 'asc'],
+    ]);
+
+    // Preserve named parameters should they exist.
+    $instance->find('all',
+        conditions: ['Articles.id' => $value],
+        order: ['Articles.id' => 'asc'],
+    );
+
+    // Array values are not spread
+    $options = ['conditions' => ['Articles.id' => $value]];
+    $instance->find('all', $options);
+}
+
+?>
+-----
+<?php
+
+namespace Cake\Upgrade\Test\TestCase\Rector\MethodCall\OptionsArrayToNamedParametersRector\Fixture;
+
+use Cake\Upgrade\Rector\Tests\Rector\MethodCall\OptionsArrayToNamedParametersRector\Source;
+
+function optionsToNamedParameters()
+{
+    $value = 1;
+    $instance = new Source\ConfigurableClass();
+
+    $instance->find();
+    $instance->find('all');
+    $instance->find('list', fields: ['name']);
+    $instance->find('all',
+    conditions: ['Articles.id' => $value],
+    order: ['Articles.id' => 'asc']);
+
+    // Preserve named parameters should they exist.
+    $instance->find('all',
+        conditions: ['Articles.id' => $value],
+        order: ['Articles.id' => 'asc'],
+    );
+
+    // Array values are not spread
+    $options = ['conditions' => ['Articles.id' => $value]];
+    $instance->find('all', $options);
```

Comment:
I wasn't sure about this but I think this is the safer assumption as we don't want to break application code.

Reply:
I think spreading the options shouldn't cause a problem but we can be safer for now. We can change it later if required after some more manual testing.